### PR TITLE
Don't allow spaces in custom flags

### LINKED ISSUE #219: [high prio] Spaces in custom contributor flags kill SPADs
If there are any spaces etc in the custom contributor flags set in here https://github.com/beyond-all-reason/teiserver/blob/0ffe7f19563cdf5498973020d24d2e26904dac7f/lib/teiserver_web/live/account/profile/contributor.ex then it kills the SPADs because teiserver sends malformed command, example log from SPADs

```
Mar 10 23:53:43 spads-eu6 spads_cluster_launcher.sh[532]: 20240310235046 - NOTICE   - [SpringLobbyInterface] Connecting to server4.beyondallreason.info:8200
Mar 10 23:53:43 spads-eu6 spads_cluster_launcher.sh[532]: 20240310235047 - WARNING  - [SpringLobbyInterface] Found invalid accountId parameter value "KOREA" (should be integer) in lobby command ADDUSER
Mar 10 23:53:43 spads-eu6 spads_cluster_launcher.sh[532]: 20240310235047 - WARNING  - [SPADS] Disconnecting from lobby server (broken connection detected)
Mar 10 23:53:43 spads-eu6 spads_cluster_launcher.sh[532]: 20240310235047 - NOTICE   - [SpringLobbyInterface] Disconnecting from server4.beyondallreason.info:8200
```

### PR DESCRIPTION
**Pull request by L-e-x-o-n**
Fixes #219

### CODE REVIEW

File: lib/teiserver_web/live/account/profile/contributor.ex [hunk 1]
```diff
@@ -73,15 +74,21 @@ defmodule TeiserverWeb.Account.ProfileLive.Contributor do
...
-    Account.update_user_stat(assigns.user.id, %{
-      "bar_plus.flag" => assigns.temp_country_code
-    })
-
-    Account.recache_user(assigns.user.id)
-
-    {:noreply,
-     socket
-     |> assign(:country_code, assigns.temp_country_code)}
+    if String.match?(assigns.temp_country_code, ~r/^\S+$/) do
```
Comment (p2004a) [inline lib/teiserver_web/live/account/profile/contributor.ex:77]: Too broad. Better something strict like `^[a-zA-Z0-9-]{2,8}$`, could be maybe even more strict then that.

Comment (L-e-x-o-n) [reply lib/teiserver_web/live/account/profile/contributor.ex]: Updated regex to you suggestion. Not sure how to go more strict than that while allowing existing flags.

File: lib/teiserver_web/live/account/profile/contributor.ex [hunk 2]
```diff
@@ -69,19 +71,27 @@ defmodule TeiserverWeb.Account.ProfileLive.Contributor do
...
-    Account.update_user_stat(assigns.user.id, %{
-      "bar_plus.flag" => assigns.temp_country_code
-    })
-
-    Account.recache_user(assigns.user.id)
-
-    {:noreply,
-     socket
-     |> assign(:country_code, assigns.temp_country_code)}
+    if String.match?(assigns.temp_country_code, ~r/^[a-zA-Z0-9-]{2,8}$/) do
```
Comment (StanczakDominik) [inline lib/teiserver_web/live/account/profile/contributor.ex:79]: Note to self: verify that this passes with all the flags in the chobby repo.

**Review by StanczakDominik — APPROVED** [review]
Looks good, I'll run one check later when I'm at PC and it should be okay to merge :)

File: lib/teiserver_web/live/account/profile/contributor.ex (hunk 2 above)
Comment (Beherith) [reply lib/teiserver_web/live/account/profile/contributor.ex:79]: One does not:
`europeanunion.png`

Comment (Beherith) [reply lib/teiserver_web/live/account/profile/contributor.ex:79]: Please allow up to 16 characters in the regex, to be sure

Comment (L-e-x-o-n) [reply lib/teiserver_web/live/account/profile/contributor.ex:79]: `eu.png` already exists for EU, `europeanunion` is a duplicate.

Comment (L-e-x-o-n) [reply lib/teiserver_web/live/account/profile/contributor.ex:79]: Increased the size anyway

**Pull request merged**